src/test/run-pass/issue-54716.rs

```diff
@@ -38,49 +38,81 @@ impl Drop for D {
 }
 
 /// Check that unused bindings are dropped after the function is polled.
-async fn foo(x: D, _y: D) {
+async fn foo_async(x: D, _y: D) {
+    x.1.borrow_mut().push(DropOrder::Function);
+}
+
+fn foo_sync(x: D, _y: D) {
     x.1.borrow_mut().push(DropOrder::Function);
 }
 
 /// Check that underscore patterns are dropped after the function is polled.
-async fn bar(x: D, _: D) {
+async fn bar_async(x: D, _: D) {
+    x.1.borrow_mut().push(DropOrder::Function);
+}
+
+fn bar_sync(x: D, _: D) {
     x.1.borrow_mut().push(DropOrder::Function);
 }
```

Comment:
pre-existing, but I think we should reserve tests named `issue-123.rs` for simple reproduction tests, random ICEs, and the like. For this test, which is more of a thorough "spec test", can we rename this to something like

```
async-await/drop-order-for-async-fn-parameters.rs
```

and add a comment like 

---

Test that the drop order for parameters in a `fn` and `async fn`  matches up. Also test that parameters (used or unused) are not dropped until the async fn completes execution. See also #54716.

Reply:
Fixed.